### PR TITLE
[Concept entry] Kotlin Objects.md

### PR DESCRIPTION
**Pull request by diosDeNada**
### Description

-Creat an entry concept for Kotlin object declaration.

### Type of Change

- Adding a new entry

### Checklist

- [ x] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:3]:
```suggestion
Description: 'Kotlin object declaration is a special type of declaration that allows the creation of a single instance of a class, commonly known as a singleton.'
```
Description is a description of the term, not the entry.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:15]:
```suggestion
Kotlin **`object`** declaration is a special type of declaration that allows the creation of a single instance of a class, commonly known as a singleton.
```
Note, we avoid first and second person pronouns in Docs.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:18]:
```suggestion
```

This is not how to declare pseudocode.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:19]:
````suggestion

```pseudo
````
Code blocks in markdown are set off with triple backticks.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:31]:
````suggestion
```

````

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:20]:
```suggestion
object MyObject {
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:33]:
```suggestion

```
Replace this with a line break.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:35]:
```suggestion
## Example

```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:36]:
```suggestion
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:37]:
````suggestion
```
````

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:47]:
````suggestion
```

````

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:48]:
```suggestion
This example shows an object called `HelloKitty` that contains a method called `sayHello`. The `sayHello` method takes a `String` parameter called `name` and displays the message `Hi, name!` on the console.
```
Again, avoiding first and second person pronouns.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:49]:
```suggestion

The following is printed to the console:
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:51]:
````suggestion
```shell
Hi, Liany<3!
```
````

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@diosDeNada I did a first review and had some comments. Mostly around two points:

1) Properly formatting Markdown code blocks. (They need to be enclosed in triple backticks ` ``` `
2) There needs to be a blank line between paragraphs, headers, and code blocks.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:12]:
```suggestion
  - 'paths/computer-science'
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:7]:
```suggestion
  - 'Android'
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:8]:
```suggestion
  - 'Kotlin'
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:9]:
```suggestion
  - 'Objects'
```

Comment (diosDeNada) [reply content/kotlin/concepts/Objects/Objects.md:9]: Thanks. I gonna work on all of that right now.

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:14]:
```suggestion

```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:15]:
```suggestion

Kotlin **`object`** declaration is a special type of declaration that allows the creation of a single instance of a class, commonly known as a Singleton.
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:19]:
````suggestion
```pseudo
````

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:32]:
```suggestion
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:33]:
```suggestion

The object declaration combines the definition of the class and the creation of its instance into a concise and powerful construct.
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:35]:
```suggestion
## Example

```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:47]:
```suggestion
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:48]:
```suggestion

This example, shows an object called `HelloKitty` that contains a method called `sayHello`. The `sayHello` method takes a `String` parameter called `name` and displays the message `Hi, name!` on the console.
```

File: content/kotlin/concepts/Objects/Objects.md
Comment (SSwiniarski) [inline content/kotlin/concepts/Objects/Objects.md:49]:
```suggestion

The following is printed to the console:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
Some more comments, mostly about spacing.

**Comment by diosDeNada** [comment]
@SSwiniarski all suggestion applied :)

**Review by SSwiniarski — APPROVED** [review]
@diosDeNada Looks good for a second review.

**Comment by diosDeNada** [comment]
Cool! I will be attentive to what follows. Thank you so much.

File: content/kotlin/concepts/Objects/Objects.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/Objects/Objects.md:3]: I don't think the issue for this entry was specific regarding what should and shouldn't be addressed about objects. But, I think this entry could take a broader approach and include the other ways in which objects are used, such as object expressions and companion objects. Please revise this entry to include the other functions of objects in the kotlin language.

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@diosDeNada I've made a couple of comments. Please update the entry, and tag me when you've finished making changes. Thanks

File: content/kotlin/concepts/Objects/Objects.md
Comment (diosDeNada) [reply content/kotlin/concepts/Objects/Objects.md:3]: Understood. I'm going to expand the entry and add the other definitions (and examples) about Kotlin Objects. I hope I can do it tomorrow. Thank you so much @caupolicandiaz.

**Comment by diosDeNada** [comment]
@caupolicandiaz sorry for the delay, I've been busier than I'd like at work. Tomorrow Saturday (in Costa Rica), in the morning, I will be spending time at the entry.

**Comment by diosDeNada** [comment]
@caupolicandiaz i finally could update the entry. I remain attentive to possible corrections. Thank you.

**Review by caupolicandiaz — APPROVED** [review]
@diosDeNada For the sake of expediency I went ahead and pushed a number of additional edits in lieu of leaving comments. Please review the changes for future reference. Considering the number of changes, from the previous round, I think this entry should probably get one more review.

**Comment by diosDeNada** [comment]
@caupolicandiaz I don't quite understand the comments you are referring to. In the previous revision, I only found a comment inside the code, related to something that @SSwiniarski had already made me correct, and that in fact remained the same after this second revision. I saw the edits you made, and I appreciate that the entry is now more accurate according to Docs standards, especially grammatically, since I notice that the format did not have to be corrected as much.
This was my first collaboration on an Open Source project, and pretty much my first time using GitHub for anything other than trying to get a page.
I will be more cautious and precise when collaborating in the future. Thanks.

**Review by KTom101 — CHANGES_REQUESTED** [review]
@diosDeNada I've added a few additions.

**Review by KTom101 — APPROVED** [review]
@diosDeNada I made one additional change directly in commit [c370620](https://github.com/Codecademy/docs/pull/2328/commits/c37062095d4419feb967d15b362125b305074315). This otherwise LGTM 👍

**Comment by KTom101** [comment]
@diosDeNada Entry is now live here: https://www.codecademy.com/resources/docs/kotlin/objects

Screenshot 2023-06-26 at 3 54 07 PM

cc @yangc95

**Pull request merged**
